lib/SILOptimizer/Differentiation/PullbackCloner.cpp

```diff
@@ -2993,13 +3092,60 @@ AdjointValue PullbackCloner::Implementation::accumulateAdjointsDirect(
       }
       return makeAggregateAdjointValue(lhsVal->getType(), newElements);
     }
+    // x + (zeroBase, index, eltToAdd) => (x.0, ..., x.index + eltToAdd, ...,
```

Comment:
@rxwei @asl I looked into adding a unit test (I'm referring to actual unit tests testing C++ classes/functions directly) for this function, but ran into some issues due to the internal state changes in PullbackCloner, that happen during this its call. 

Do you guys see any issues if I create a PullbackCloner constructor initializing minimal state, required for this test?